### PR TITLE
Add service config

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

**Review by MykolaSolyanko — APPROVED** [review]

**Review by mlohvynenko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

File: src/common/tools/semver.cpp [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+        return {0, ErrorEnum::eInvalidArgument};
+    }
+
+    for (size_t i = 0; i < Min(identifiers1.Size(), identifiers2.Size()); i++) {
+        if (auto result = CompareNumericIdentifiers(identifiers1[i], identifiers2[i]);
+            !result.mError.IsNone() || result.mValue != 0) {
+            return result;
+        }
+
+        continue;
```
Comment (mykola-kobets-epam) [inline src/common/tools/semver.cpp:293]: not required

File: src/common/tools/semver.cpp [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    for (const auto& identifier : identifiers) {
+        if (auto err = ValidateStrIdentifier(identifier); !err.IsNone()) {
+            return err;
+        }
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error SplitVersion(String& version, Array<String>& parts)
```
Comment (mykola-kobets-epam) [inline src/common/tools/semver.cpp:194]: it looks the array consists of 3 elements in all the cases. 
mb better split it into 3 parameters

File: src/common/tools/semver.cpp [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+
+    if (parts1[1].IsEmpty() && !parts2[1].IsEmpty()) {
+        return {1, ErrorEnum::eNone};
+    }
+
+    if (parts1[1].IsEmpty() && parts2[1].IsEmpty()) {
+        return {0, ErrorEnum::eNone};
+    }
+
+    return ComparePrereleaseParts(parts1[1], parts2[1]);
```
Comment (mykola-kobets-epam) [inline src/common/tools/semver.cpp:431]: metadata part isn't checked

**Review by mykola-kobets-epam — APPROVED** [review]

File: src/common/tools/semver.cpp [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+ **********************************************************************************************************************/
+
+namespace {
+
+/***********************************************************************************************************************
+ * Consts
+ **********************************************************************************************************************/
+
+constexpr auto cMaxNumIdentifiers = 8;
+const char*    cAllowedChars      = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-";
```
[the remark this answers is not in the document]
Comment (al1img) [reply src/common/tools/semver.cpp:23]: No, I'm checking each identifier. Accordign to semver spec, allowed identifier symbols are [0-9A-Za-z-].

File: src/common/tools/semver.cpp (hunk 1 above)
Comment (al1img) [reply src/common/tools/semver.cpp:293]: removed

File: src/common/tools/semver.cpp (hunk 3 above)
Comment (al1img) [reply src/common/tools/semver.cpp:431]: metadata isn't taken into account for semver comparison according to spec

**Review by mlohvynenko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

**Pull request merged**